### PR TITLE
5 como cliente quiero comprar criptomonedas

### LINKED ISSUE #5: Como cliente quiero comprar criptomonedas
**Descripción**
Recibo una petición POST con el id de una cartera, el id de una moneda y la cantidad y introduzco la cantidad de esa moneda en la cartera.

**Ejemplo**

Missing coin id:
Request: POST /coin/buy
```
{
  "wallet_id": "string",
  "amount_usd": 0
}
```
Response: 400 BAD REQUEST
```
{
  "error": "coin id missing from request"
}
```

Missing wallet id:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "amount_usd": 0
}
```
Response: 400 BAD REQUEST
```
{
  "error": "wallet id missing from request"
}
```

Missing amount:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "wallet_id": "string"
}
```
Response: 400 BAD REQUEST
```
{
  "error": "amount missing from request"
}
```

Coin not found:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "wallet_id": "string",
  "amount_usd": 0
}
```
Response: 404 NOT FOUND
```
{
  "error": "coin not found for specific id"
}
```

Wallet not found:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "wallet_id": "string",
  "amount_usd": 0
}
```
Response: 404 NOT FOUND
```
{
  "error": "wallet not found for specific id"
}
```

Amount not positive:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "wallet_id": "string",
  "amount_usd": 0
}
```
Response: 400 BAD REQUEST
```
{
  "error": "amount should be positive"
}
```

Service unavailable:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "wallet_id": "string",
  "amount_usd": 0
}
```
Response: 503 SERVICE UNAVAILABLE
```
{
  "error": "service is unavailable"
}
```

Happy Path:
Request: POST /coin/buy
```
{
  "coin_id": "string",
  "wallet_id": "string",
  "amount_usd": 0
}
```
Response: 200 OK

### PR DESCRIPTION
**Pull request by jesuslezaun**
Pull request for issue #5

### CODE REVIEW

File: tests/app/Application/BuyCryptocurrencies/BuyCryptocurrenciesServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    {
+        $coinId = "1";
+        $walletId = "1";
+        $amountUsd = 0;
+
+        $this->cryptoDataSource
+            ->expects('findCoinById')
+            ->with($coinId)
+            ->once();
+
```
Comment (xAriextz) [inline tests/app/Application/BuyCryptocurrencies/BuyCryptocurrenciesServiceTest.php]: Delete empty line

File: tests/app/Infrastructure/Controller/BuyCryptocurrenciesControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    /**
+     * @test
+     */
+    public function amountNotPositive()
+    {
+        $this->cryptoDataSource
+            ->expects('findCoinById')
+            ->with('999')
+            ->once();
+
```
Comment (xAriextz) [inline tests/app/Infrastructure/Controller/BuyCryptocurrenciesControllerTest.php]: Delete empty line

File: tests/app/Infrastructure/Controller/BuyCryptocurrenciesControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        $coin = new Coin("90", "Bitcoin", "BTC", 0, 6010);
+        $wallet = new Wallet("1");
+
+        $this->cryptoDataSource
+            ->expects('findCoinById')
+            ->with('90')
+            ->once()
+            ->andReturn($coin);
+
```
Comment (xAriextz) [inline tests/app/Infrastructure/Controller/BuyCryptocurrenciesControllerTest.php]: Delete empty lines

**Review by raulsc-cl — APPROVED** [review]
Okey to merge

**Review by xAriextz — CHANGES_REQUESTED** [review]
Consider making changes based on my feedback

**Review by xAriextz — APPROVED** [review]
Ok to merge

**Review by raulsc-cl — APPROVED** [review]
Okey to merge

**Pull request merged**